modules/scylladb/scylladb_test.go

```diff
@@ -0,0 +1,288 @@
+		require.Equal(t, "Amazing ScyllaDB Test", clusterName)
+	})
+}
+
+func TestScyllaAlternator(t *testing.T) {
+	ctx := context.Background()
+
+	alternatorPort := uint16(8000)
+
+	t.Run("test-with-alternator", func(t *testing.T) {
+		ctr, err := scylladb.Run(ctx,
+			"scylladb/scylla:6.2.2",
+			scylladb.WithAlternator(alternatorPort),
+		)
+		testcontainers.CleanupContainer(t, ctr)
+		require.NoError(t, err)
+
+		cli, err := getDynamoAlternatorClient(t, ctr, alternatorPort)
+		require.NoError(t, err)
+		requireCreateTable(t, cli)
+	})
+
+	t.Run("test-without-alternator", func(t *testing.T) {
+		ctr, err := scylladb.Run(ctx,
+			"scylladb/scylla:6.2",
+		)
+		testcontainers.CleanupContainer(t, ctr)
+		require.NoError(t, err)
+
+		cli, err := getDynamoAlternatorClient(t, ctr, alternatorPort)
+		require.Error(t, err)
+		require.Nil(t, cli)
+	})
+}
+
+// scyllaAlternatorResolver is a custom endpoint resolver for the ScyllaDB Alternator.
+type scyllaAlternatorResolver struct {
+	HostPort string
+}
+
+// ResolveEndpoint resolves the endpoint for the ScyllaDB Alternator.
+func (r *scyllaAlternatorResolver) ResolveEndpoint(_ context.Context, _ dynamodb.EndpointParameters) (smithyendpoints.Endpoint, error) {
+	return smithyendpoints.Endpoint{
+		URI: url.URL{Host: r.HostPort, Scheme: "http"},
+	}, nil
+}
+
+// getDynamoAlternatorClient returns a new DynamoDB client for the ScyllaDB Alternator.
+func getDynamoAlternatorClient(t *testing.T, c *scylladb.Container, port uint16) (*dynamodb.Client, error) {
+	t.Helper()
+
+	hostPort, err := c.ConnectionHost(context.Background(), port)
+	if err != nil {
+		return nil, fmt.Errorf("connection host: %w", err)
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithCredentialsProvider(credentials.StaticCredentialsProvider{
+		Value: aws.Credentials{
+			AccessKeyID:     "SCYLLA_ALTERNATOR_ACCESS_KEY_ID",
+			SecretAccessKey: "SCYLLA_ALTERNATOR_SECRET_ACCESS",
+		},
+	}))
+	if err != nil {
+		return nil, fmt.Errorf("load default config: %w", err)
+	}
+
+	return dynamodb.NewFromConfig(cfg, dynamodb.WithEndpointResolverV2(&scyllaAlternatorResolver{HostPort: hostPort})), nil
+}
+
+// requireCreateTable creates a table in the ScyllaDB Alternator, failing the test if an error occurs.
+func requireCreateTable(t *testing.T, client *dynamodb.Client) {
+	t.Helper()
+
+	_, err := client.CreateTable(context.Background(), &dynamodb.CreateTableInput{
+		TableName: aws.String("demo_table"),
+		KeySchema: []types.KeySchemaElement{
+			{
+				AttributeName: aws.String("id"),
+				KeyType:       types.KeyTypeHash,
+			},
+		},
+		AttributeDefinitions: []types.AttributeDefinition{
+			{
+				AttributeName: aws.String("id"),
+				AttributeType: types.ScalarAttributeTypeS,
+			},
+		},
+		BillingMode: types.BillingModePayPerRequest,
+	})
+	require.NoError(t, err)
+}
+
+func TestWithCustomCommands(t *testing.T) {
+	t.Run("invalid-flag", func(t *testing.T) {
+		t.Parallel()
```

Comment:
question: looks like there's quite a bit of memory in use if these all run in parallel, could that be counter productive

Reply:
Indeed, the container request is big in memory. I added the parallel execution because they are unit tests that happens really fast even without parallel execution, so I guess it's fine to remove that